ui/components/ui/update-nickname-popover/update-nickname-popover.js

```diff
@@ -0,0 +1,119 @@
+import React, { useCallback, useContext, useState } from 'react';
+import PropTypes from 'prop-types';
+
+import Popover from '../popover';
+import Button from '../button';
+import TextField from '../text-field';
+
+import { I18nContext } from '../../../contexts/i18n';
+
+import Identicon from '../identicon/identicon.component';
+
+export default function UpdateNicknamePopover({
+  nickname,
+  address,
+  onAdd,
+  memo,
+  onClose,
+}) {
+  const t = useContext(I18nContext);
+
+  const [nicknameInput, setNicknameInput] = useState(nickname);
+  const [memoInput, setMemoInput] = useState(memo);
+
+  const handleNicknameChange = (event) => {
+    setNicknameInput(event.target.value);
+  };
+
+  const handleMemoChange = (event) => {
+    setMemoInput(event.target.value);
+  };
+
+  const closePopover = useCallback(() => {
+    onClose();
+  }, [onClose]);
+
+  const onCancel = () => {
+    onClose();
+  };
+
+  const onSubmit = () => {
+    onAdd(address, nicknameInput, memoInput);
+    onClose();
+  };
+
+  let title = t('addANickname');
```

Comment:
Can we do `const title = nickname ? t('editAddressNickname') : t('addANickname')`?